extensions/amp-access/0.1/amp-access.js

```diff
@@ -293,9 +293,8 @@ export class AccessService {
 
   /** @private */
   startInternal_() {
-    // TODO(dvoytenko, #3742): This will refer to the ampdoc once AccessService
-    // is migrated to ampdoc as well.
-    Services.actionServiceForDoc(this.ampdoc).installActionHandler(
+    const actionService = Services.actionServiceForDoc(this.accessElement_);
```

Comment:
Does this actually work in shadow mode? I just realized that `accessElement` is actually very unusual - it's not in the doc's `body`, but in `head`. There could be some differences in this regard.

Reply:
`node.ownerDocument` is not an active path in the shadow mode however.

Reply:
Sorry, got mixed up -- I meant that the FIE path is the same. 

For PWA, it follows the `AmpDocService.getAmpDoc()` code path which walks up the shadow root: https://github.com/ampproject/amphtml/blob/master/src/service/ampdoc-impl.js#L180